FinSearchUnified/Subscriber/Widgets.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace FinSearchUnified\Subscriber;
+
+use Enlight\Event\SubscriberInterface;
+
+class Widgets implements SubscriberInterface
+{
+    /**
+     * Returns an array of event names this subscriber wants to listen to.
+     *
+     * @return array The event names to listen to
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            'Shopware_Controllers_Widgets_Listing::listingCountAction::before' => 'beforeListingCountAction'
+        ];
+    }
+
+    public function beforeListingCountAction(\Enlight_Hook_HookArgs $args)
+    {
+        /** @var \Shopware_Controllers_Widgets_Listing $subject */
+        $subject = $args->getSubject();
+
+        $params = $subject->Request()->getParams();
+
+        if (!isset($params['sSearch']) && !isset($params['sCategory'])) {
```

Comment:
I didn't consider this before but `\FinSearchUnified\Helper\StaticHelper::useShopSearch` should also be checked. So at the beginning of the method, if `useShopSearch` is `true` you can return without any further computation.

Adapt the test `testBeforeListingCountAction` so that `useShopSearch` returns false and implement a separate test which checks the behavior for:
* `useShopSearch` -> `false`
* `sSearch` isn't present
* `sCategory` isn't present

Reply:
@ihanli how will this be implemented? if findologic search is active, then it will always have sSearch present if neither category or search is provided. Or did you mean useShopSearch -> true

Reply:
Yes, you are right. I updated the previous comment.